src/drop.test-d.ts

```diff
@@ -0,0 +1,46 @@
+import { drop } from "./drop";
+import { pipe } from "./pipe";
+
+describe("data first", () => {
+  it("works on regular inputs", () => {
+    expectTypeOf(drop([1, 2, 3, 4, 5], 2)).toEqualTypeOf<[3, 4, 5]>();
+  });
+
+  it("works on empty arrays", () => {
+    expectTypeOf(drop([], 2)).toEqualTypeOf<[]>();
+  });
+
+  it("works with negative numbers", () => {
+    expectTypeOf(drop([1, 2, 3, 4, 5], -1)).toEqualTypeOf<[5]>();
+  });
+
+  it("works when dropping more than the length of the array", () => {
+    expectTypeOf(drop([1, 2, 3, 4, 5], 10)).toEqualTypeOf<[]>();
+  });
+
+  it("works with generalized typed arrays", () => {
+    expectTypeOf(drop([] as Array<number>, 2)).toEqualTypeOf<[]>();
+  });
+
+  it("works with generalized typed count", () => {
+    expectTypeOf(drop([1, 2, 3, 4, 5], 2 as number)).toEqualTypeOf<never>();
```

Comment:
**question:** can we change the return type to avoid this case?

we can still base our type off `ArraySlice<T, N, T["length"]>`, maybe something like check if `N` is literal, use the arrayslice, otherwise return generic type

Reply:
Good idea, done :)

The whole `pipe` issue is solvable BTW
we'll need to add `pipe<const A, B, ...>` to the signature, but it breaks other functions that use it, so I'm not sure what's the best way to handle this